### PR TITLE
Add a registration requirements field

### LINKED ISSUE #2179: Add a specific "registration-requirements" field to competitions
One of the changes in the new Regulations release will be to make mandatory the announcement of the requirements that a competitor has to fulfill in order to complete their registration. To do so, as suggested by @viroulep, we think that a specific field in the competition form makes sense. Additionally, this field would have to be locked once the competition is confirmed. Please, see thewca/wca-regulations#489.

### LINKED ISSUE #2478: Allow adding custom refund policy message on the registration page
Currently, the registration page just lists the price of registration.

Many competitions do not allow refunds.  CubingUSA Nationals only allows 50% refunds.

This is important information that needs to be present on the registration page so warn people properly.

(Related to issue #2179 on displaying registration requirements, since that should also be on the registration page.)

### PR DESCRIPTION
**Pull request by viroulep**
Fixes #2179.
Fixes #2478.

The WRC want to enforce transparency regarding registration requirements at competitions.
This introduces a new field for registration requirements, which can be modified by organizers/delegates only before the competition is confirmed.

### CODE REVIEW

**Comment by viroulep** [comment]
This should only be merged when the new Regulations are published, as thewca/wca-regulations#489 is not yet in `official`.

File: WcaOnRails/config/locales/en.yml [hunk 1]
```diff
@@ -384,6 +385,7 @@ en:
         competitor_limit_reason: "What is the reason for the limit on competitors? Please fill this out in English!"
         venueAddress: ""
         championship_type: ""
+        registration_requirements: "Please indicate here any specifities about the registration process, as needed per <a href=\"https://www.worldcubeassociation.org/regulations/guidelines.html#2c1+\" target=\"_blank\">Guideline 2c1+</a>. After the competition is announced, this field will <b>not</b> be editable anymore, and Board's approval will be needed to modify it.<br/>For example indicate here if competitors need to have a clean payment history with a specific entity, or if they will have to answer a confirmation email in case the maximum number of registrations is reached."
```
Comment (viroulep) [inline WcaOnRails/config/locales/en.yml]: @thewca/wrc-team for review on the wording.

Comment (Laura-O) [reply WcaOnRails/config/locales/en.yml]: Looks good!

**Comment by pedrosino** [comment]
Registration procedure may change as the competition gets closer (maybe you
will allow on-site registration or disallow it), so I guess those would be
requested to the Board. Is that right?

Em 20/11/2017 08:26, "Laura Ohrndorf" <notifications@github.com> escreveu:

**Comment by Laura-O** [comment]
> Registration procedure may change as the competition gets closer (maybe you
> will allow on-site registration or disallow it), so I guess those would be
> requested to the Board. Is that right?

I am not sure if you understood "registration procedure" correctly.
This could be prepayment for registration approval or replying to an email sent a week before the competition.

And yes, the description text already is "... this field will <b>not</b> be editable anymore, and Board's approval will be needed to modify it"

File: WcaOnRails/config/locales/en.yml (hunk 1 above)
Comment (jfly) [reply WcaOnRails/config/locales/en.yml]: Is "specifities" a word?

Also, just checking, but does html work here?

Comment (viroulep) [reply WcaOnRails/config/locales/en.yml]: In Frenglish it definitely is, but it looks like it [does exist](http://www.wordreference.com/enfr/specificity) in English, and is a synonym of "peculiarities".

Although it does mean what I wanted, I'll just switch this to "all details" if it sounds unusual.

Comment (viroulep) [reply WcaOnRails/config/locales/en.yml]: Oh wow, I can't even read properly my own word, and obviously meant "specificities" :p

Comment (jfly) [reply WcaOnRails/config/locales/en.yml]: Hahaha, Frenglish. I think "all details" or "any details" sounds more natural to me =)

Comment (jfly) [reply WcaOnRails/config/locales/en.yml]: Oh, or "specificities"! That's a fine word.

Comment (viroulep) [reply WcaOnRails/config/locales/en.yml]: Yep, I just fixed the typo.

**Comment by viroulep** [comment]
Actually the field's hint refers to a Guideline that doesn't exist yet, so I'll rebase this and merge it when we release the new version of the Regulations :)

**Comment by viroulep** [comment]
It's been overdue for one month now, I've just rebased it and will merge it soon.

**Comment by viroulep** [comment]
I realized there was another concern about this being displayed on the "Register" page, so I implemented it.
Here is a screenshot:
reg_requirements

It's displayed even if the registrations are not open or if the user is not logged in.

File: WcaOnRails/config/locales/en.yml [hunk 2]
```diff
@@ -383,6 +384,7 @@ en:
         competitor_limit_reason: "What is the reason for the limit on competitors? Please fill this out in English!"
         venueAddress: ""
         championship_type: ""
+        registration_requirements: "Please indicate here any specificities about the registration process, as needed per <a href=\"https://www.worldcubeassociation.org/regulations/guidelines.html#2c1+\" target=\"_blank\">Guideline 2c1+</a>. After the competition is announced, this field will <b>not</b> be editable anymore, and Board's approval will be needed to modify it.<br/>For example indicate here if competitors need to have a clean payment history with a specific entity, or if they will have to answer a confirmation email in case the maximum number of registrations is reached."
```
Comment (jfly) [inline WcaOnRails/config/locales/en.yml]: Does this render correctly? Shouldn't this be called `registration_requirements_html`?

It also might be nice to mention here that the registration requirements will also show up on the register page, which raises the question of what language this should be written in. I guess you should tell people to fill this out both in English, and in the language your competitors speak?

Comment (viroulep) [reply WcaOnRails/config/locales/en.yml]: It does render correctly actually.
The examples [here](http://guides.rubyonrails.org/i18n.html#using-safe-html-translations) only include explicit use through `t`, and here it's a hint for a field, maybe that's why.

**Comment by viroulep** [comment]
Quick ping here: are you fine with how the registrations requirements are displayed?

File: WcaOnRails/config/locales/en.yml [hunk 3]
```diff
@@ -383,6 +384,7 @@ en:
         competitor_limit_reason: "What is the reason for the limit on competitors? Please fill this out in English!"
         venueAddress: ""
         championship_type: ""
+        registration_requirements: "Please indicate here any specificities about the registration process, as needed per <a href=\"https://www.worldcubeassociation.org/regulations/guidelines.html#2c1+\" target=\"_blank\">Guideline 2c1+</a>. After the competition is announced, this field will <b>not</b> be editable anymore, and Board's approval will be needed to modify it.<br/>For example indicate here if competitors need to have a clean payment history with a specific entity, if there is some refund policy, or if they will have to answer a confirmation email in case the maximum number of registrations is reached.<br/>Please fill these information <b>at least in English</b>; it will be displayed on the competition's registration page if you are using the WCA website registrations."
```
Comment (jfly) [inline WcaOnRails/config/locales/en.yml]: "Please fill these information *at least in English*" feels awkward to me. Maybe something like this instead?

"Please fill this out in English and whatever other languages your competitors speak"

**Pull request merged**